### PR TITLE
Submitting Week 1 Gene Finder

### PR DESCRIPTION
**Pull request by srbarden**
Submitting my work on the functions in gene_finder.py for week 1

### CODE REVIEW

File: gene_finder.py [hunk 1]
```diff
@@ -45,15 +51,39 @@ def get_reverse_complement(dna):
...
     'TGAACGCGG'
     """
-    # TODO: implement this
-    pass
+    rev_comp = ''
+    for i in range(0, len(dna)):
+        nucleo = dna[i]
+        comp = get_complement(nucleo)
+        rev_comp = comp + rev_comp
+    return rev_comp
```
Comment (segerphilip) [inline gene_finder.py]: Good. Another way would be with `dna = dna[::-1]`, a python function that gets reverse of lists/strings.

File: gene_finder.py [hunk 2]
```diff
@@ -78,9 +115,22 @@ def find_all_ORFs_oneframe(dna):
...
+    all_ORFs_oneframe = []
+    length_of_orf = 0
+    while i < len(dna_split):
+        if dna_split[i] == "ATG":
+            orf = rest_of_ORF(dna[int(i)*3:])
+            all_ORFs_oneframe += [orf]
+            length_of_orf = math.ceil(len(orf)/3)
+            i += length_of_orf
+        else:
+            i += 1
```
Comment (segerphilip) [inline gene_finder.py]: Probably want to increment by 3 (codon length), however I think you work around this because of split_into_codons, so nevermind 😕

**Comment by segerphilip** [comment]
Where is your work for week 2/completing Gene Finder?

File: gene_finder.py [hunk 3]
```diff
@@ -45,15 +52,39 @@ def get_reverse_complement(dna):
...
+    >>> split_into_codons("ATGTGATAGCC")
+    ['ATG', 'TGA', 'TAG', 'CC']
+    """
+    dna_split = []
+    length = math.ceil(len(dna)/3)
+    for i in range(0, length):
+        j = 3*i
+        codon = dna[j:j+3]
+        dna_split += [codon]
+    return dna_split
```
Comment (SungwooPark) [inline gene_finder.py]: Good use of a helper function here.

File: gene_finder.py [hunk 4]
```diff
@@ -96,8 +147,11 @@ def find_all_ORFs(dna):
     >>> find_all_ORFs("ATGCATGAATGTAG")
     ['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG']
     """
-    # TODO: implement this
-    pass
+    all_ORFs = []
+    for i in range(0, 3):
```
Comment (SungwooPark) [inline gene_finder.py]: A tip: if you give only one parameter to range function, the function assumes that the lower bound is 0. For example `range(0,3)` and `range(3)` is effectively same thing.

File: gene_finder.py [hunk 5]
```diff
@@ -109,8 +163,9 @@ def find_all_ORFs_both_strands(dna):
     >>> find_all_ORFs_both_strands("ATGCGAATGTAGCATCAAA")
     ['ATGCGAATG', 'ATGCTACATTCGCAT']
     """
-    # TODO: implement this
-    pass
+    orfs = []
```
Comment (SungwooPark) [inline gene_finder.py]: I don't think this initialization of a list is necessary here.

File: gene_finder.py [hunk 6]
```diff
@@ -119,8 +174,9 @@ def longest_ORF(dna):
     >>> longest_ORF("ATGCGAATGTAGCATCAAA")
     'ATGCTACATTCGCAT'
     """
-    # TODO: implement this
-    pass
+    orfs = find_all_ORFs_both_strands(dna)
+    longest = max(orfs, key=len)
```
Comment (SungwooPark) [inline gene_finder.py]: Good use of max function. Make sure you understand what `key=len` represents here.

File: gene_finder.py [hunk 7]
```diff
@@ -158,9 +229,24 @@ def gene_finder(dna):
...
+    aa_sequences = []
+    i = 0
+    while i < len(orfs):
+        print(len(orfs[i]))
+        if len(orfs[i]) > threshold:
+            print('if')
+            aa_sequences += [coding_strand_to_AA(orfs[i])]
+        i += 1
+    print(aa_sequences)
+
```
Comment (SungwooPark) [inline gene_finder.py]: You probably want to return `aa_sequences` instead of printing.

**Comment by SungwooPark** [comment]
Most of the functions look good but there is a bug in your program that does not exclude stop codon when the full dna is loaded. Take a look at comments for some suggestions.

**Pull request closed without merging**